src/hoi4_world/roles/stories_creator.cpp

```diff
@@ -1,13 +1,58 @@
 #include "src/hoi4_world/roles/stories_creator.h"
 
+#include <ranges>
+
 #include "external/fmt/include/fmt/format.h"
 #include "src/hoi4_world/roles/roles_importer.h"
 
 
 
-void hoi4::CreateStories()
+namespace
+{
+
+using Tag = std::string;
+using CombinationName = std::string;
+
+
+std::vector<std::pair<Tag, CombinationName>> MakeCombinations(const std::map<std::string, hoi4::Role>& roles,
+    const std::map<std::string, hoi4::Country>& countries)
+{
+   std::vector<std::pair<Tag, CombinationName>> combinations;
+
+   for (const auto& [role_name, role]: roles)
+   {
+      // scan for 'tag=TAG' constructs
+      std::regex tag_match_regex(R"([\s\S]+tag=(.+)[\s\S]+)");
+      std::smatch match;
+      if (!std::regex_match(role.GetRequirements(), match, tag_match_regex))
+      {
+         continue;
+      }
+      std::string required_tag = match[1];
+
+      for (const std::string& country_tag: countries | std::views::keys)
```

Comment:
Do you really need to iterate over all countries for every pass? It's a map, would .contains() suffice?

Reply:
Left an explanatory comment (we'll eventually have more complex conditions)